lacework/resource_lacework_alert_channel_datadog.go

```diff
@@ -111,80 +111,70 @@ func resourceLaceworkAlertChannelDatadogCreate(d *schema.ResourceData, meta inte
 
 	var (
 		lacework = meta.(*api.Client)
-		datadog  = api.NewDatadogAlertChannel(d.Get("name").(string),
-			api.DatadogChannelData{
-				DatadogSite:    site,
-				DatadogService: service,
-				ApiKey:         d.Get("api_key").(string),
+		datadog  = api.NewAlertChannel(d.Get("name").(string),
+			api.DatadogAlertChannelType,
+			api.DatadogDataV2{
+				DatadogSite: string(site),
```

Comment:
We wouldn't need to cast to string again if we set the DatadogSite and Type as `datadogSite`  and `datadogService` in the go-sdk
https://github.com/lacework/go-sdk/pull/570/files#r723033234

An alternative would be to change datadog_site and datadog_type in the schema. But I think having the protection for only valid types would be preferable.